u2f-ref-code/java/src/com/google/u2f/codec/RawMessageCodec.java

```diff
@@ -215,4 +291,65 @@ public static AuthenticateResponse decodeAuthenticateResponse(byte[] data) throw
     .put(challengeSha256);
     return signedData;
   }
+  
+  public static byte[] encodeTransferAccessMessageSignedBytesForAuthenticationKey(
+      byte sequenceNumber, byte[] newUserPublicKey, byte[] applicationSha256,
+      X509Certificate attestationCertificate) throws U2FException {
+
+    byte[] attestationCertificateBytes;
+    try {
+      attestationCertificateBytes = attestationCertificate.getEncoded();
+    } catch (CertificateEncodingException e) {
+      throw new U2FException("Error when encoding attestation certificate.", e);
+    }
+
+    byte[] signedData = new byte[1 + newUserPublicKey.length + applicationSha256.length
+        + attestationCertificateBytes.length];
+    
+    ByteBuffer.wrap(signedData)
```

Comment:
Also, I believe this would actually have to look like this:

ByteArrayOutputStream signedDataOutputStream = new ByteArrayOutputStream();
signedDataOutputStream.write(sequenceNumber);
signedDataOutputStream.write(newUserPublicKey);
...
signedDataOutputStream.write(signatureUsingAuthenticationKey)

return signedDataOutputStream.toByteArray();

This is because there is no "put" method, just write, which returns void. You can't keep writing to void.